sunpy/map/mapbase.py

```diff
@@ -193,7 +193,28 @@ def __init_subclass__(cls, **kwargs):
         if cls.__doc__ is None:
             # Set an empty string, to prevent an error adding None to str in the next line
             cls.__doc__ = ''
-        cls.__doc__ += textwrap.indent(_notes_doc, "    ")
+        existing_notes_pos = cls.__doc__.find('Notes\n    -----')
+        existing_notes_pos2 = _notes_doc.find('Notes\n-----')
+        existing_notes_data = textwrap.indent(_notes_doc[existing_notes_pos2 + len('Notes\n-----'):].strip(), "    ")
+        references_pattern = "References\n    ----------"
+        examples_pattern = "Examples\n   -------"
+        start_index = existing_notes_pos + len('Notes\n    -----')
+        references_pos = cls.__doc__.find(references_pattern, start_index)
+        examples_pos = cls.__doc__.find(examples_pattern, start_index)
+        if existing_notes_pos != -1:
+            if references_pos != -1 or examples_pos != -1:
+                next_pattern_pos = min(pos for pos in [references_pos, examples_pos] if pos != -1)
+                other_patterns = cls.__doc__[:next_pattern_pos]
+                cls.__doc__ = (other_patterns + existing_notes_data.lstrip() + '\n    ' + cls.__doc__[next_pattern_pos:])
+            else:
+                cls.__doc__ += existing_notes_data
+        else:
+            if references_pos != -1 or examples_pos != -1:
+                next_pattern_pos = min(pos for pos in [references_pos, examples_pos] if pos != -1)
+                other_patterns = cls.__doc__[:next_pattern_pos]
+                cls.__doc__ = (other_patterns + 'Notes\n    -----\n' + existing_notes_data + '\n    ' + cls.__doc__[next_pattern_pos:])
+            else:
+                cls.__doc__ += textwrap.indent(_notes_doc, "    ")
```

Comment:
```suggestion
    existing_notes_pos = cls.__doc__.find('Notes\n    -----')
    existing_notes_pos2 = _notes_doc.find('Notes\n-----')
    if existing_notes_pos2 != -1:
        existing_notes_data = textwrap.indent(_notes_doc[existing_notes_pos2 + len('Notes\n-----'):].strip(), "    ")
    else:
        existing_notes_data = ''
    references_pattern = "References\n    ----------"
    examples_pattern = "Examples\n   -------"
    start_index = existing_notes_pos + len('Notes\n    -----')
    references_pos = cls.__doc__.find(references_pattern, start_index)
    examples_pos = cls.__doc__.find(examples_pattern, start_index)
    next_pattern_pos = min(pos for pos in [references_pos, examples_pos] if pos != -1)
    if existing_notes_pos != -1:
        cls.__doc__ = cls.__doc__[:next_pattern_pos] + existing_notes_data + '\n    ' + cls.__doc__[next_pattern_pos:]
    else:
        cls.__doc__ += 'Notes\n    -----\n' + existing_notes_data
```
Can you see if this suggestion works?

Reply:
This does work but we also want to make sure that the notes heading is above references and examples, which is not being solved by this.

Reply:
So in fact it doesn't work?

Reply:
yeah!